sympy/geometry/util.py

```diff
@@ -657,14 +654,21 @@ def intersection(*entities):
     Reals should be converted to Rationals, e.g. Rational(str(real_num))
     or else failures due to floating point issues may result.
 
+    Case 1: When the keyword argument 'pairwise' is False (default value):
+    In this case, the functon returns a list of the points
+    through which all the entities pass.
+
+    Case 2: When the keyword argument 'pairwise' is True:
+    In this case, the functions returns a list of all points
+    where intersections take place.(Please Note that these points
+    may not necessarily lie on all the entities)
+
     See Also
     ========
-
     sympy.geometry.entity.GeometryEntity.intersection
 
     Examples
     ========
-
     >>> from sympy.geometry import Point, Line, Circle, intersection
     >>> p1, p2, p3 = Point(0, 0), Point(1, 1), Point(-1, 5)
```

Comment:
Let's make this example really simple to visualize and use the same example for both cases:

```
>>> from sympy import Cirlc, Line, Point, oo
>>> origin = Point(0, 0)
>>> c = Circle(origin, radius=1).translate(y=1)
>>> xaxis = Line(origin, slope=0)
>>> yaxis = Line(origin, slope=oo)
>>> intersection(c, xaxis, yaxis)
Point2D(0, 0)
>>> intersection(c, xaxis, yaxis, pairwise=True)
[Point2D(0, 0), Point2D(0, 2)]
```